### ISSUE TITLE
Allow ConversationId to be customized

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus#4171]
When sending commands that relates to the same "business entity" its useful to override the `ConversationId` to use the ID of the business entity since that will show all message interactions for the entity in the same view in tools like ServiceInsight.

Eg.

```
var options = new SendOptions();

options.SetConversationId(orderId);

context.Send(new PlaceOrderCommand{ ... }, options);
```

**Comment by indualagarsamy** [comment]
@andreasohlund - updates on this?

**Comment by andreasohlund** [comment]
Since we do support setting the conversation id via headers this doesn't seem urgent? (still a valid feature though)

**Comment by ramonsmits** [comment]
Made doco changes for this https://github.com/Particular/docs.particular.net/pull/3447

**Comment by timbussmann** [comment]
@andreasohlund https://github.com/Particular/NServiceBus/issues/4683 sounds like a duplicate to me?

**Comment by andreasohlund** [comment]
This was shipped in 7.4 https://docs.particular.net/nservicebus/messaging/headers#messaging-interaction-headers-nservicebus-conversationid-starting-a-new-conversation
